named tuple in closure argument destructuring breaks after formatting

When I have a file with named tuple destructuring in closure argument like this:

```
❯ cat bug1.swift
    let dict: [String: String] = ["a":"b"]

    dict.forEach { (header: (key: String, value: String)) in
        print(header.key)
        print(header.value)
    }
❯ swiftc bug1.swift && ./bug1
a
b
```

and I try to format it using

```
❯ swiftformat --version
0.53.10
```

without any configuration file

```
❯ swiftformat bug1.swift
Running SwiftFormat...
warning: No Swift version was specified, so some formatting features were disabled. Specify the version of Swift you are using with the --swiftversion option, or by adding a .swift-version file to your project.
SwiftFormat completed in 0.02s.
1/1 files formatted.
```

I get this error when compiling

```
❯ swiftc bug1.swift && ./bug1
bug1.swift:4:18: error: value of type 'String' has no member 'key'
    print(header.key)
          ~~~~~~ ^~~
bug1.swift:5:18: error: value of type 'String' has no member 'value'
    print(header.value)
          ~~~~~~ ^~~~~
error: fatalError
```

because the code formats to this

```
❯ cat bug1.swift
  let dict: [String: String] = ["a": "b"]

  for (header, value) in dict {
      print(header.key)
      print(header.value)
  }
```

and realistically should either format to this:

```
  let dict: [String: String] = ["a": "b"]

  for (key, value) in dict {
      print(key)
      print(value)
  }
```

or not be changed at all.

--- a/Sources/ParsingHelpers.swift
+++ b/Sources/ParsingHelpers.swift
@@ -1860,8 +1860,19 @@ extension Formatter {
                 return (argumentNames: [], inKeywordIndex: inKeywordIndex)
             }
 
-            let argumentTokens = tokens[firstTokenInArgumentsList ... endOfArgumentsScopeIndex].split(separator: .delimiter(","))
-            argumentNames = argumentTokens.compactMap { $0.first(where: \.isIdentifier)?.string ?? $0[0].string }
+            // Parse the comma-separated list of arguments
+            var currentArgIndex = firstTokenInArgumentsList
+            while tokens[currentArgIndex].isIdentifierOrKeyword {
+                argumentNames.append(tokens[currentArgIndex].string)
+
+                guard let nextComma = index(of: .delimiter(","), in: currentArgIndex ..< endOfArgumentsScopeIndex),
+                      let nextArgLabel = index(of: .identifierOrKeyword, in: nextComma ..< endOfArgumentsScopeIndex)
+                else {
+                    break
+                }
+
+                currentArgIndex = nextArgLabel
+            }
         }
 
         // Otherwise this is an anonymous closure

--- a/Tests/RulesTests+Syntax.swift
+++ b/Tests/RulesTests+Syntax.swift
@@ -4896,6 +4896,28 @@ class SyntaxTests: RulesTests {
         testFormatting(for: input, rule: FormatRules.preferForLoop)
     }
 
+    func testForEachOverDictionary() {
+        let input = """
+        let dict = ["a": "b"]
+
+        dict.forEach { (header: (key: String, value: String)) in
+            print(header.key)
+            print(header.value)
+        }
+        """
+
+        let output = """
+        let dict = ["a": "b"]
+
+        for header in dict {
+            print(header.key)
+            print(header.value)
+        }
+        """
+
+        testFormatting(for: input, output, rule: FormatRules.preferForLoop)
+    }
+
     // MARK: propertyType
 
     func testConvertsExplicitTypeToInferredType() {

